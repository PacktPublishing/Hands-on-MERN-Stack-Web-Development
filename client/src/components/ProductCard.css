.ProductCard {
    display: flex;
    flex-direction: column;
    margin-bottom: 2rem;
}

.ProductCard img {
    width: 400px;
}

.ProductCard h3 {
    font-family: Monaco, monospace;
}

.ProductCard p {
    font-family: "Courier New", monospace;
}

.ProductCard h3,
.ProductCard p {
    margin: 0 0 0.5rem 0;
}