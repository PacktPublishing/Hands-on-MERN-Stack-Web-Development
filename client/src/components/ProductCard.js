import React, { Component } from 'react';
import './ProductCard.css';

export default class ProductCard extends Component {
  constructor(props) {
    super(props);
    this.state = {
      image: this.props.images[0],
    };
  }

  handleMouseOver = () => {
    if (this.props.images.length > 1) {
      this.setState({ image: this.props.images[1] });
    }
  };

  handleMouseLeave = () => {
    this.setState({ image: this.props.images[0] });
  };

  render() {
    return (
      <div className="ProductCard">
        <img
          src={this.state.image}
          onMouseOver={this.handleMouseOver}
          onMouseLeave={this.handleMouseLeave}
        />
        <h3>{this.props.name}</h3>
        <p>{this.props.price}</p>
      </div>
    )
  }
};
