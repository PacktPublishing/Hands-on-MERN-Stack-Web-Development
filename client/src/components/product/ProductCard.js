import React, { Component } from 'react';
import PropTypes from 'prop-types';
import './ProductCard.css';
import { SecondaryButton } from '../Button';
import TopProductLabel from './TopProductLabel';

export default class ProductCard extends Component {
  constructor(props) {
    super(props);
    this.state = {
      image: this.props.images[0],
    };
  }

  handleMouseOver = () => {
    if (this.props.images.length > 1) {
      this.setState({ image: this.props.images[1] });
    }
  };

  handleMouseLeave = () => {
    this.setState({ image: this.props.images[0] });
  };

  render() {
    return (
      <div className="ProductCard">
        <img
          src={this.state.image}
          onMouseOver={this.handleMouseOver}
          onMouseLeave={this.handleMouseLeave}
        />
        <h3>{this.props.name}</h3>
        <p>
          {this.props.price}
          {this.props.isFeatured && <TopProductLabel />}
        </p>
        {this.props.withRemoveButton &&
          <SecondaryButton onClick={this.props.onRemove}>
            Remove
          </SecondaryButton>
        }
      </div>
    )
  }
};

ProductCard.propTypes = {
  name: PropTypes.string.isRequired,
  price: PropTypes.string.isRequired,
  images: PropTypes.arrayOf(PropTypes.string),
  withRemoveButton: PropTypes.bool,
  onRemove: PropTypes.func,
};
