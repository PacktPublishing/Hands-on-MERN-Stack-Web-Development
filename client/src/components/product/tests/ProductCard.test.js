import React from 'react';
import renderer from 'react-test-renderer';
import ProductCard from '../ProductCard';

test('To match snapshot', () => {
  const output = renderer.create(
    <ProductCard
      name="Product XYZ"
      price="$49.99"
      images={['https://example.com/some.png']}
    />
  );
  expect(output.toJSON()).toMatchSnapshot();
});

test('To match snapshot when there is a remove button', () => {
  const output = renderer.create(
    <ProductCard
      name="Product XYZ"
      price="$49.99"
      images={['https://example.com/some.png']}
      withRemoveButton={true}
      onRemove={() => {}}
    />
  );
  expect(output.toJSON()).toMatchSnapshot();
});
