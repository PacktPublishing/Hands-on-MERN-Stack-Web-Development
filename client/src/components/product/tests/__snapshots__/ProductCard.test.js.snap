// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`To match snapshot 1`] = `
<div
  className="ProductCard"
>
  <img
    onMouseLeave={[Function]}
    onMouseOver={[Function]}
    src="https://example.com/some.png"
  />
  <h3>
    Product XYZ
  </h3>
  <p>
    $49.99
  </p>
</div>
`;

exports[`To match snapshot when there is a remove button 1`] = `
<div
  className="ProductCard"
>
  <img
    onMouseLeave={[Function]}
    onMouseOver={[Function]}
    src="https://example.com/some.png"
  />
  <h3>
    Product XYZ
  </h3>
  <p>
    $49.99
  </p>
  <button
    className="BaseButton SecondaryButton"
    onClick={[Function]}
  >
    Remove
  </button>
</div>
`;
